Libs/MRML/Core/vtkMRMLTextStorageNode.cxx

```diff
@@ -200,9 +200,14 @@ int vtkMRMLTextStorageNode::WriteDataInternal(vtkMRMLNode * refNode)
 
   std::ofstream file;
   file.open(fullName);
+  if (!file.is_open())
+  {
+    vtkErrorMacro("WriteData: Could not open file for writing");
```

Comment:
This is an error that we want to report to the users as well, so we should use `vtkErrorToMessageCollectionMacro` (that logs the error and also adds it to the user message collection). See for example in https://github.com/Slicer/Slicer/blob/main/Libs/MRML/Core/vtkMRMLModelStorageNode.cxx

Reply:
Done!